TLSharp.Core/Network/MtProtoSender.cs

```diff
@@ -162,11 +202,14 @@ private bool processMessage(ulong messageId, int sequence, BinaryReader messageR
             // TODO: check sessionid
             // TODO: check seqno
 
+
             //logger.debug("processMessage: msg_id {0}, sequence {1}, data {2}", BitConverter.ToString(((MemoryStream)messageReader.BaseStream).GetBuffer(), (int) messageReader.BaseStream.Position, (int) (messageReader.BaseStream.Length - messageReader.BaseStream.Position)).Replace("-","").ToLower());
             needConfirmation.Add(messageId);
+            Ack().Wait();
```

Comment:
shouldn't this use `await Ack();`?

Reply:
Error: The 'await' operator can only be used within an async method. 
I will return it back, sorry limited by time

Reply:
then make processMessage be async

Reply:
Then here is 5 methods in a row

Reply:
sure?

Reply:
the method Ack does not need to be async as the method it calls Send does not have any await calls itself.  Should these still be marked async ?